packages/text-input/src/TextInput/TextInput.spec.tsx

```diff
@@ -105,95 +102,21 @@ describe('packages/text-input', () => {
     spy.mockClear();
   });
 
-  describe('when the "state" is "valid"', () => {
-    test('displays checkmark icon when input is valid', () => {
-      const { getInputValue, isValid, isOptional } = renderTextInput({
-        value: validEmail,
-        state: State.Valid,
-        optional: true,
-        ...defaultProps,
-      });
-
-      expect(getInputValue()).toBe(validEmail);
-      expect(isValid()).toBe(true);
-      expect(isOptional()).toBe(false);
-    });
-
-    test('displays checkmark icon when input is valid even when input is disabled', () => {
-      const { getInputValue, isOptional } = renderTextInput({
-        value: validEmail,
-        state: State.Valid,
-        disabled: true,
-        ...defaultProps,
-      });
-
-      expect(getInputValue()).toBe(validEmail);
-      expect(isOptional()).toBe(false);
-    });
-  });
-
-  describe('disabled', () => {
-    test('is true', () => {
-      const { isDisabled } = renderTextInput({ disabled: true });
-
-      expect(isDisabled()).toBe(true);
-    });
-
-    test('is false', () => {
-      const { isDisabled } = renderTextInput();
-
-      expect(isDisabled()).toBe(false);
-    });
+  test('does not render "optional" text when the prop is set to false', () => {
+    const { container } = renderTextInput({ optional: false, ...defaultProps });
+    expect(container.innerHTML).not.toContain('Optional');
   });
 
-  describe('when the "state" is "error"', () => {
-    test('displays warning icon when input is invalid', () => {
-      const { isOptional, getInputValue, isError } = renderTextInput({
-        value: invalidEmail,
-        state: State.Error,
-        optional: true,
-        ...defaultProps,
-      });
-
-      expect(getInputValue()).toBe(invalidEmail);
-      expect(isError()).toBe(true);
-      expect(isOptional()).toBe(false);
-    });
-
-    test('renders empty error message container', () => {
-      const { getErrorMessage } = renderTextInput({
-        value: invalidEmail,
-        state: State.Error,
-        optional: true,
-        ...defaultProps,
-      });
-      expect(getErrorMessage()).toHaveTextContent('');
-    });
-
-    test('displays error message when input is invalid', () => {
-      const { getErrorMessage } = renderTextInput({
-        value: invalidEmail,
-        state: State.Error,
-        optional: true,
-        errorMessage: error,
-        ...defaultProps,
-      });
-      expect(getErrorMessage()).toHaveTextContent(error);
+  test(`renders with aria-disabled attribute but not disabled attribute when disabled prop is set`, () => {
+    const { textInput } = renderTextInput({
+      disabled: true,
     });
+    expect(textInput.getAttribute('aria-disabled')).toBeTruthy();
+    expect(textInput.getAttribute('disabled')).toBeFalsy();
   });
 
   describe('when the "state" is "none"', () => {
     defaultProps.onChange.mockReturnValue('none');
-
```

Comment:
Why don't we need this test?

Reply:
I removed tests that should be tested in `FormField`